File: app/src/main/java/no/nav/tilbakemeldingsmottak/rest/feilogmangler/validation/MeldFeilOgManglerValidator.java
package no.nav.tilbakemeldingsmottak.rest.feilogmangler.validation;

import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator;
import no.nav.tilbakemeldingsmottak.rest.feilogmangler.domain.MeldFeilOgManglerRequest;
import org.springframework.stereotype.Component;

@Component
public class MeldFeilOgManglerValidator extends RequestValidator {

    public void validateRequest(MeldFeilOgManglerRequest request) {
        isNotNull(request.getOnskerKontakt(), "onskerKontakt");
        if (request.getOnskerKontakt()) {
            hasText(request.getEpost(), "epost", " dersom onskerKontakt=true");
        }
        isNotNull(request.getFeiltype(), "feiltype");
        hasText(request.getMelding(), "melding");
    }

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/rest/feilogmangler/validation/MeldFeilOgManglerValidator.kt
package no.nav.tilbakemeldingsmottak.rest.feilogmangler.validation

import no.nav.tilbakemeldingsmottak.model.MeldFeilOgManglerRequest
import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator
import org.springframework.stereotype.Component

@Component
class MeldFeilOgManglerValidator : RequestValidator() {
    fun validateRequest(request: MeldFeilOgManglerRequest) {
        isNotNull(request.onskerKontakt, "onskerKontakt")
        if (request.onskerKontakt == true) {
            hasText(request.epost, "epost", " dersom onskerKontakt=true")
            isLegalEmail(request.epost, "epost", " dersom epost")
        }
        isNotNull(request.feiltype, "feiltype")
        hasText(request.melding, "melding")
        maxSize(request.melding, 10000, "melding")
    }
}
